app/controllers/projects_controller.rb

```diff
@@ -163,10 +163,16 @@ def update
   # DELETE /projects/1
   # DELETE /projects/1.json
   def destroy
-    @project.destroy
+    deletable = @project.samples.to_a.empty?
```

Comment:
I disabled project.samples so this might always be true. 
Please do Sample.where(project_id: @project.id).count == 0